ClientApp/viking-vault/src/UserDashboard/CreateAutomaticPaymentModal.tsx

```diff
@@ -0,0 +1,258 @@
+            }
+        return  {
+            message: "A company must be selected!",
+            class: "alert alert-info"
+        }
+    }
+
+    private amountMustBePositive = () => {
+        if(this.state.automaticPayment.amount.match(regexCheckIfPositiveFloat))
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "The amount must be a positive number!",
+            class: "alert alert-info"
+        }
+    }
+
+    private initialPaymentDateIsAfterCurrentDate = () => {
+        if(this.state.automaticPayment.initialPaymentDate !== "" && new Date(this.state.automaticPayment.initialPaymentDate) >= currentDate)
+            return  {
+                message: "Ok",
+                class: "alert alert-success"
+            }
+        return  {
+            message: "Initial Payment Date must be after the current date!",
+            class: "alert alert-info"
+        }
+    }
+
+    private mandatoryFieldsCompletedCorrectly = () => {
+        let val = false;
+        const validationFunctions = [this.isCompanySelected, this.amountMustBePositive, this.initialPaymentDateIsAfterCurrentDate];
+        val = validationFunctions.every(functionItem => functionItem().message === "Ok");
+        return val;
+    }
+
+    private resetInputValues = () => {
+        this.setState({
+            automaticPayment: {
+                ...this.state.automaticPayment,
+                companyName: "",
+                amount: "",
+                initialPaymentDate: "",
+            }
+        });
+    }
+
+    private closeModal = () => {
+        this.resetInputValues();
+        this.props.onModalClose(false);
+    }
+
+    private getIdOfCompanyByName = (name: string) => {
+        return this.state.companies.filter(company => company.name === name)[0].id;
+    }
+
+    private getAutomaticPayment = () => {
+        return {
+            companyId: this.getIdOfCompanyByName(this.state.automaticPayment.companyName),
+            companyName: this.state.automaticPayment.companyName,
+            amount: this.state.automaticPayment.amount,
+            initialPaymentDate: new Date(this.state.automaticPayment.initialPaymentDate),
+            lastPaymentDate: new Date(2000,0,2),
+        };
+    }
+
+    getAllCompanies = () => {
+        var companies;
+        var token = sessionStorage.getItem("Authentication-Token");
+        if(token === null)
+        {
+            this.setState({
+                errorLabel: "Access Token Unavailable",
+            })
+            return [];
+        }
+        else{
+            fetch(constants.baseUrl+"company", {
+                method: "GET",
+                headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json',
+                'x-access-token' : token.toString()
+                }})
+            .then( response => 
+                {
```

Comment:
JS convention: curly bracket on the same line
`blah {`

`}`

Reply:
is this resolved? I don't see it as resolved